### PR TITLE
Add ad-hoc query introduction to basics-part1 and materialized docs

### PR DESCRIPTION
**Pull request by Karakatiza666**
Fix https://github.com/feldera/feldera/issues/2712: Missing tutorial section on populating tables.
Fix https://github.com/feldera/feldera/issues/2768: Materialized views documentation is outdated

### CODE REVIEW

File: docs/tutorials/basics/part1.md [hunk 1]
```diff
@@ -100,42 +100,104 @@ of **changes** to the view, but cannot inspect its current contents.
...
+```sql
+INSERT INTO PRICE (part, vendor, price) VALUES
+(1, 2, 10000),
+(2, 1, 15000),
+(3, 3, 9000);
+```
+
+The ad-hoc queries we run are executed by the same pipeline that handles the stream processing, but instead of using our incremental computation engine they are evaluated against the latest snapshot of the pipeline's tables and views.
+
+Enough with the static SQL, let's see Feldera's incremental computation in action!
```
Comment (ryzhyk) [inline docs/tutorials/basics/part1.md:146]: From here on, I think we could be more pedagogical. 

* Don't send the user switching between change stream and sql tabs. Let's first explain incremental view maintenance by using SQL only (you have most of this text below, so this is mostly changing the structure).
  * Explain that whenever you insert into a table, Feldera incrementally updates all vies
  * Direct the user to `SELECT *` from one of the views.
  * Modify one of the inputs and `SELECT *` again to see the modified view

* Introduce the change stream:
  * explain that in Feldera you can observe a stream of changes to the views, not just the current snapshot
  * Switch to the change stream tab, tick the boxes.
  * Switch back to SQL, modify one of the tables again
  * Observe deltas in the Change stream
(BTW, having to switch between the tabs is probably bad ergonomics; we should think about a UI that combines the two views on the same screen)

* Inserting JSON data directly in the change stream. I suggest we don't go there in this tutorial.  The focus here is on understanding the basics, and this feature isn't really needed for that.

File: docs/tutorials/basics/part1.md [hunk 2]
```diff
@@ -100,42 +100,104 @@ of **changes** to the view, but cannot inspect its current contents.
...
+
+![Inserted vendor data](basics-part1-2.png)
+
+Yep, everything is in order. Let's fill the other table:
+
+```sql
+INSERT INTO PART (id, name) VALUES
+(1, 'Flux Capacitor'),
+(2, 'Warp Core'),
+(3, 'Kyber Crystal')
```
Comment (ryzhyk) [inline docs/tutorials/basics/part1.md:134]: Please add semicolons to all queries. We don't require them now, but we may in the future, and it's the standard behavior for SQL CLIs.

**Review by ryzhyk — CHANGES_REQUESTED** [review]
Looks great, the main suggestion is to structure the incremental compute section (Step 3) a bit better.

File: docs/tutorials/basics/part1.md (hunk 2 above)
Comment (Karakatiza666) [reply docs/tutorials/basics/part1.md:134]: Left them out on purpose, but will change to include them if that's the case

**Review by ryzhyk — APPROVED** [review]

**Comment by gz** [comment]
fyi currently getting


```
ERROR in ./feldera/docs/tutorials/basics/part3.md
Module build failed (from ./node_modules/@docusaurus/mdx-loader/lib/index.js):
Error: MDX compilation failed for file "/home/gz/workspace/docs.feldera.com/feldera/docs/tutorials/basics/part3.md"
Cause: Image feldera/docs/tutorials/basics/supply-chain-analytics.png used in feldera/docs/tutorials/basics/part3.md not found.
Details:
Error: Image feldera/docs/tutorials/basics/supply-chain-analytics.png used in feldera/docs/tutorials/basics/part3.md not found.
    at ensureImageFileExist (/home/gz/workspace/docs.feldera.com/node_modules/@docusaurus/mdx-loader/lib/remark/transformImage/index.js:85:15)
    at async getImageAbsolutePath (/home/gz/workspace/docs.feldera.com/node_modules/@docusaurus/mdx-loader/lib/remark/transformImage/index.js:107:5)
    at async processImageNode (/home/gz/workspace/docs.feldera.com/node_modules/@docusaurus/mdx-loader/lib/remark/transformImage/index.js:126:23)
    at async Promise.all (index 0)
    at async /home/gz/workspace/docs.feldera.com/node_modules/@docusaurus/mdx-loader/lib/remark/transformImage/index.js:139:9
    at Object.mdxLoader (/home/gz/workspace/docs.feldera.com/node_modules/@docusaurus/mdx-loader/lib/loader.js:130:25)
```

make sure to run yarn build in docs.feldera.com repo with this branch before merging

File: docs/sql/materialized.md [hunk 3]
```diff
@@ -29,21 +29,42 @@ These declarations instruct Feldera to maintain a complete snapshot of the table
 
 ## Using materialized tables and views
 
-### Web Console
+You can explore the contents of materialized tables and views by issuing ad-hoc queries. Unlike the SQL programs that define a pipeline and are evaluated incrementally, ad-hoc queries are evaluated against the current snapshot of pipeline's tables and views. Ad-hoc queries can be executed both when the pipeline is running or paused.
```
[the remark this answers is not in the document]
Comment (mihaibudiu) [reply docs/sql/materialized.md:32]: we have to make it 100% clear that these queries use a different dialect of SQL, which we don't support, and may give different results than Feldera's SQL. This mechanism is only here to allow people to explore the data, not to use on a regular basis - unless they are happy with the semantic differences.

**Comment by Karakatiza666** [comment]
> make sure to run yarn build in docs.feldera.com repo with this branch before merging

Yeah, sorry, I just left this step for the end

**Pull request merged**